pkg/event/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,90 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/pkg/event"
+)
+
+// FromKubernetesEvent converts a Kubernetes Kargo event to a CloudEvent.
+func FromKubernetesEvent(evt corev1.Event) (event.Meta, error) {
+	var parsedEvent event.Meta
+	var err error
+
+	switch kargoapi.EventType(evt.Reason) {
```

Comment:
This feels like something we're going to forget to update. I don't have any bright ideas though.

Reply:
Yep, I have (or had) a grumpy comment about "my kingdom for a real enum". The only thing I've found that might be helpful is this linter: https://github.com/nishanths/exhaustive